gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageImplTest.java

```diff
@@ -141,7 +142,55 @@ public void writeLargeObject_withSmallUploadChunk() throws IOException {
             resourceId,
             /* partitionSize= */ 5 * 1024 * 1024,
             partitionsCount);
+    assertObjectContent(helperGcs, resourceId, partition, partitionsCount);
+    assertThat(trackingGcs.requestsTracker.getAllRequestStrings())
+        .containsExactlyElementsIn(
+            getExpectedRequestsForCreateObject(
+                resourceId, uploadChunkSize, partitionsCount, partition))
+        .inOrder();
+  }
+
+  @Test
+  public void writeLargeObject_withSmallUploadChunk_testStatistics() throws IOException {
+    StorageResourceId resourceId = new StorageResourceId(TEST_BUCKET, name.getMethodName());
+
+    int uploadChunkSize = 1024 * 1024;
+    TrackingStorageWrapper<GoogleCloudStorageImpl> trackingGcs =
+        newTrackingGoogleCloudStorage(getOptionsWithUploadChunk(uploadChunkSize));
 
+    int partitionsCount = 1;
+    byte[] partition =
+        writeObject(
+            trackingGcs.delegate,
+            resourceId,
+            /* partitionSize= */ 5 * 1024 * 1024,
+            partitionsCount);
+    List<String> expectedRequests =
+        getExpectedRequestsForCreateObject(resourceId, uploadChunkSize, partitionsCount, partition);
+    assertThat(
+            trackingGcs
+                .delegate
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST)
+                .longValue())
+        .isEqualTo(getCountFromExpectedRequests("GET", expectedRequests));
+    assertThat(
+            trackingGcs
+                .delegate
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST_FAILURES)
+                .longValue())
+        .isEqualTo(1L);
+    assertThat(
+            trackingGcs
+                .delegate
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_PUT_REQUEST)
+                .longValue())
+        .isEqualTo(getCountFromExpectedRequests("PUT", expectedRequests));
+    assertThat(
+            trackingGcs
+                .delegate
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_PUT_REQUEST_FAILURES)
+                .longValue())
+        .isEqualTo(4L);
     assertObjectContent(helperGcs, resourceId, partition, partitionsCount);
```

Comment:
This triggers a read, please add assertions on metrics after this call as well

Reply:
The values have not changed ! 

Please use `trackingGcs.delegate` to read and assert the right values

Reply:
Updated